### PR TITLE
RD-5288 cypress commands

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: https://cloudifysource.atlassian.net/browse/RD-5288

## Screenshots / Videos
N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
`blueprints_terraform_spec` passed: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3306/pipeline

`auth_spec`: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3315/pipeline

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3190/pipeline

 

## Documentation
N/A

### CODE REVIEW

**Comment by leon-good-life** [comment]
I started system tests: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3190/pipeline

**Comment by kubama** [comment]
I'm not sure about the change in terms of **labels** commands. My points are:
- these commands are used only in `deployment_actions_spec.ts` yet they are being made globally available commands
- these commands make sense only in very specific context which makes their reuse difficult and actually unlikely - as such they are not good candidates for global commands

In the end I think it somehow violates best practices: https://docs.cypress.io/api/cypress-api/custom-commands#1-Don-t-make-everything-a-custom-command

@PatrycjaKubiczek, @qooban - do you have any thoughts on this?

**Comment by qooban** [comment]
@kubama I checked it and seems to me like these commands are used not only in `deployment_actions_spec.ts`, but also in `labels_spec.ts`. The one exception is `typeLabelInput` which is used only in `deployment_actions_spec.ts`. In that case - what do you think about moving `typeLabelInput` to `deployment_actions_spec.ts`?

**Comment by kubama** [comment]
> @kubama I checked it and seems to me like these commands are used not only in `deployment_actions_spec.ts`, but also in `labels_spec.ts`.

Looks like I missed it as `labels_spec.ts` is not being updated in this PR.

> The one exception is `typeLabelInput` which is used only in `deployment_actions_spec.ts`. In that case - what do you think about moving `typeLabelInput` to `deployment_actions_spec.ts`?

Actually I can see only `addLabel` being used in `labels_spec.ts`. So I would move all except `addLabel` to `deployment_actions_spec.ts`.

**Comment by qooban** [comment]
> So I would move all except `addLabel` to `deployment_actions_spec.ts`.

Agree, let's do it.

**Comment by leon-good-life** [comment]
> > So I would move all except `addLabel` to `deployment_actions_spec.ts`.
> 
> Agree, let's do it.

Other functions in `test/cypress/support/labels.ts`  (`typeLabelKey` and `typeLabelValue` functions) are called inside `addLabel`. So it is difficult to move them to `deployment_actions_spec.ts`.

**Comment by qooban** [comment]
> Other functions in `test/cypress/support/labels.ts` (`typeLabelKey` and `typeLabelValue` functions) are called inside `addLabel`. So it is difficult to move them to `deployment_actions_spec.ts`.

So seems like my [original idea](https://github.com/cloudify-cosmo/cloudify-stage/pull/2080#issuecomment-1191465795) make sense 🙂 We can move `typeLabelInput` to `deployment_actions_spec.ts` and the rest should stay in `test/cypress/support/labels.ts`.

File: test/cypress/integration/widgets/deployment_actions_spec.ts [hunk 1]
```diff
@@ -111,7 +110,7 @@ describe('Deployment Action Buttons widget', () => {
 
         it('adds new label by typing', () => {
             cy.get('.modal').within(() => {
-                addLabel('sample_key', 'sample_value');
+                cy.addLabel('sample_key', 'sample_value');
```
Comment (leon-good-life) [inline test/cypress/integration/widgets/deployment_actions_spec.ts:113]: I don't understand why the `cy.addLabel` doesn't work:
Screen Shot 2022-07-27 at 21 32 24

**Comment by leon-good-life** [comment]
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3292/pipeline

**Review by qooban — COMMENTED** [review]
I see errors like "resource_commons_1.waitUntilNotEmpty  is not a function" coming from `test/cypress/integration/widgets/blueprints_terraform_spec.ts`. Once you fix it, I can do the review.

**Comment by leon-good-life** [comment]
I started `blueprints_terraform_spec`: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3306/pipeline

One test in `blueprints_terraform_spec` was also not passing in master branch.

**Comment by leon-good-life** [comment]
`blueprints_terraform_spec` passed

**Review by qooban — APPROVED** [review]

**Pull request merged**